### ISSUE TITLE
How can I use moment-timezone with 10 year data range with Typescript types

### ISSUE THREAD

**Issue by garethdn** [moment/moment-timezone#1089]
If I import moment timezone with

`import * as moment from 'moment-timezone';`

I get all the required typings but all the data is preloaded. If I require only the 10 year range and I import like so

`import moment from "moment-timezone/builds/moment-timezone-with-data-10-year-range"`

I get only the data I want but no typings. Likewise if I import only the library with no data, I get no typings

`import moment from 'moment-timezone/moment-timezone `

**Comment by gilmoreorless** [comment]
You're right — we currently only have types defined for `moment-timezone` and `moment-timezone/moment-timezone-utils`, but not any of the other import paths. I'll take a look at adding some extra definition files.

**Comment by LuisWolf-Snapaddy** [comment]
Is there any update on this?

Maybe you can use the [`exports` field of the package.json](https://docs.npmjs.com/cli/v11/configuring-npm/package-json#exports) like this to explicitly export the pre-built JS files via a separate entry point. Since it's the same JS but just bundled you should be able to specify the same `index.d.ts` file for the types. 🤔 

![Image](https://github.com/user-attachments/assets/b82a5df6-bbc0-4e4b-8235-cb70aeea1208)

**Comment by gilmoreorless** [comment]
Apologies, it's been a while since an update. I did start looking at exporting all the variations of the types, then I found several problems with the existing types and went down a rabbit hole. Then I had some fairly major health concerns and this kind of slipped through the cracks.

I'll try to prioritise fixing up the types soon. Though it likely won't be by using the `exports` field as suggested — I know from experience in other projects that just adding that field causes all sorts of breaking changes for existing consumers.

**Comment by gilmoreorless** [comment]
I've just released [version `0.6.0`](https://github.com/moment/moment-timezone/releases/tag/0.6.0) with a bunch of TypeScript-only fixes, including definitions for the generated files.

Try it out and let me know if there are any problems with the new types.

**Comment by sslincoco** [comment]
您的邮件已收到,我会尽快给您回复。
